packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -988,7 +988,28 @@ export async function getLogs<ReturnFormat extends DataFormat>(
 	filter: Filter,
 	returnFormat: ReturnFormat,
 ) {
-	const response = await ethRpcMethods.getLogs(web3Context.requestManager, filter);
+	let { toBlock, fromBlock } = filter;
+
+	if (toBlock) {
+		if (typeof toBlock === 'bigint') {
+			toBlock = `0x${toBlock.toString(16)}`;
+		}
+		if (typeof toBlock === 'number') {
+			toBlock = numberToHex(toBlock);
+		}
+	}
+	if (fromBlock) {
+		if (typeof fromBlock === 'bigint') {
+			fromBlock = `0x${fromBlock.toString(16)}`;
+		}
+		if (typeof fromBlock === 'number') {
+			fromBlock = numberToHex(fromBlock);
+		}
+	}
```

Comment:
Should use [toHex](https://github.com/web3/web3.js/blob/51a59f952f1d535be864b0c85966ecc571b451b3/packages/web3-utils/src/converters.ts#L335)

Reply:
Also instead of doing `if (toBlock/fromBlock)` use `isNullish` to remain consistent with the rest of the library

Reply:
addressed